### ISSUE TITLE
No IAGeofence Id iOS sdk

### ISSUE THREAD

**Issue by NirmanKalia** [IndoorAtlas/ios-sdk-swift-examples#33]
Hi,

Unlike Android sdk the iOS sdk does not have a `getId()` or `getName()` function for identifying a geofence. I could only find  the following fields:-
```
  boundingBox
  floor
  payload
  isCloudGeofence
– containsCoordinate:
```
which I don’t  think can be used for a unique id parameter (except `boundingBox.coords` but that is not ideal ).

Please tell me if there is a way to get the id or name and If I’m missing something 

Thank You,
Nirman

**Comment by matti-ida** [comment]
Hi Nirman,

geofence identifier and name are available via the IARegion interface which IAGeofence inherits from.

Thanks.

**Comment by NirmanKalia** [comment]
> Hi Nirman,
> 
> geofence identifier and name are available via the IARegion interface which IAGeofence inherits from.
> 
> Thanks.

Hi @matti-ida ,

I tried getting the name from IAregion object from `indoorLocationManager(_ manager: IALocationManager, didUpdateLocations locations: [Any])` and `indoorLocationManager(_ manager: IALocationManager, didEnter region: IARegion)` but none of them gave the geofence name or id even when the current location was inside the Geofence. 
Instead It showed the Venue and floor name.

Thanks

**Comment by matti-ida** [comment]
Hi,

in didEnterRegion, you need to check the IARegion.type whether it's a geofence: https://docs.indooratlas.com/ios/latest/constants/ia_region_type

in swift: `region.type == .iaRegionTypeGeofence`

The objective-c examples include a geofence example: https://github.com/IndoorAtlas/ios-sdk-examples/blob/master/example/src/Examples/GeofenceViewController.m#L45

**Comment by NirmanKalia** [comment]
Hi @matti-ida ,

I already tried that even though the location is inside the geofence (verified that through the `region.venue?.geofences` list and the current location returned true for one of them `geofencesListItem.contains(locations.last.location!.coordinate)` ) but still `region.type` is not `. iaRegionTypeGeofence ` how can that be?

Thanks.

**Comment by matti-ida** [comment]
Hi,

there will be separate enter/exit callbacks for each region type.

**Comment by matti-ida** [comment]
Also the floor number of a geofence must match the floor number of current location for the geofence to trigger.

**Comment by NirmanKalia** [comment]
Hi @matti-ida ,
There is just one floor in my App, and the location shows to be on that floor. I understand  that there will be separate callbacks. but those callbacks are not happening even the location is inside the Geofence.

Thanks

**Comment by NirmanKalia** [comment]
Should I raise another issue for this

**Comment by Cloudef** [comment]
Are you using the cloud geofences or dynamic ones (created in the app runtime)? If the latter, can you show how you are creating the geofence.

**Comment by NirmanKalia** [comment]
No, I am using Cloud Geofences. There are multiple Geofences in the floor and none of them trigger a Callback.

**Comment by matti-ida** [comment]
Which SDK version are you using? There was a fix for cloud geofences in SDK 3.1.4:
https://docs.indooratlas.com/technical/changelogs/ios/

The fix is also included in recently released 3.2.1 GA.
